packages/shared/src/components/notifications/NotificationItem.spec.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import NotificationItem, { NotificationItemProps } from './NotificationItem';
+import {
+  NotificationAvatarType,
+  NotificationType,
+} from '../../graphql/notifications';
+
+const sampleNotificationTitle = 'Welcome to your new notification center!';
+const sampleNotificationDescription =
+  'The notification system notifies you of important events such as replies, mentions, updates etc.';
+const sampleNotification: NotificationItemProps = {
+  isUnread: true,
+  icon: 'icon link',
+  title: `<p>${sampleNotificationTitle}</p>`,
+  description: `<p>${sampleNotificationDescription}</p>`,
+  type: NotificationType.System,
+  targetUrl: 'post url',
+  attachments: [
+    { title: 'Sample attachment', image: 'sample attachment image' },
+  ],
+  avatars: [
+    {
+      name: 'source',
+      image: 'source avatar',
+      referenceId: 'source',
+      targetUrl: 'webapp/source',
+      type: NotificationAvatarType.Source,
+    },
+    {
+      name: 'user',
+      image: 'user avatar',
+      referenceId: 'user',
+      targetUrl: 'webapp/user',
+      type: NotificationAvatarType.User,
+    },
+  ],
+};
+
+describe('notification attachment', () => {
+  it('should display image', async () => {
+    const [attachment] = sampleNotification.attachments;
+    render(<NotificationItem {...sampleNotification} />);
+    const img = await screen.findByAltText(
+      `Preview cover of the article: ${attachment.title}`,
+    );
+    expect(img).toHaveAttribute('src', attachment.image);
+  });
+
+  it('should have a title', async () => {
+    const [attachment] = sampleNotification.attachments;
+    render(<NotificationItem {...sampleNotification} />);
+    await screen.findByText(attachment.title);
+  });
+});
+
+describe('notification avatars', () => {
+  it('should display the avatar of the source', async () => {
+    const [source] = sampleNotification.avatars;
+    render(<NotificationItem {...sampleNotification} />);
+    const img = await screen.findByAltText(`${source.referenceId}'s profile`);
+    expect(img).toHaveAttribute('src', source.image);
+  });
+
+  it('should display the avatar of the user', async () => {
+    const [, user] = sampleNotification.avatars;
+    render(<NotificationItem {...sampleNotification} />);
+    const img = await screen.findByAltText(`${user.referenceId}'s profile`);
+    expect(img).toHaveAttribute('src', user.image);
+  });
+
+  it('should not display anything if the type is unknown', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const notif = { ...sampleNotification } as any;
```

Comment:
Should it not cast as `Notification` or something?

Reply:
This is the part where the type of notification is not found in our system. I used any so we can put values outside of the current `enum NotificationType`.